docs/production-candig.md

```diff
@@ -0,0 +1,122 @@
+# Guide to CanDIG production deployments
+
+Apart from the basic steps in the [CanDIGv2 Install Guide](install-candig.md) to get the candig stack up and running, there are additional settings and security recommendations that need to set up in a production level environment. We provide the following as general advice, but it is important for all CanDIG deployers to also consult with their institutional infrastructure security personnel to ensure that their deployment meets the necessary level of data security.
+
+## Reverse Proxy & Firewall
+
+It is essential to setup a reverse proxy and firewall so that only specific ports are open to the internet. The software used for this is up to the deployer and is considered outside of the CanDIG stack. 
+
+Basically, the only ports that should be available are to tyk (443) and keycloak (80).
+
+Some specific examples of how existing institutes have approached this are below.
+
+### HAProxy - UHN & BCGSC
+
+At UHN, the candig.uhnresearch.ca domain is under a proxy, so requests to a specific service go through the following stack:
+
+```
+[ CLIENT ] ---> | HTTPS | ---> [ UHN PROXY ] ---> | HTTP | ---> [ CANDIG_PROD / TYK ] ---> | HTTP | ---> [ CANDIG_DATA_SERVICES ]
+```
+
+Specifically, the UHN proxy forwards all candig.uhnresearch.ca and candigauth.uhnresearch.ca requests (all ports) to candig1:5080 (tyk) and candig1:8080 (keycloak) respectively, thereby acting as a firewall.  All CanDIGv2 microservices can only be accessed through Tyk.
```

Comment:
It technically only forwards on https connections -- conenctions on port 80 redirect to 443, and it otherwise doesn't respond to most other ports.
```suggestion
Specifically, the UHN proxy forwards all candig.uhnresearch.ca and candigauth.uhnresearch.ca requests (port 443) to candig1:5080 (tyk) and candig1:8080 (keycloak) respectively, thereby acting as a firewall.  All CanDIGv2 microservices can only be accessed through Tyk.
```